share/functions/export.fish

```diff
@@ -10,7 +10,7 @@ function export --description 'Set global variable. Alias for set -gx, made for
                 set -gx $v $$v
             case 2
                 if contains $v[1] PATH CDPATH MANPATH
-                    set -gx $v[1] (string split ":" $v[2])
+                    set -gx $v[1] (string split ":" (string replace " " ":" $v[2]))
```

Comment:
Exactly. The problem here is that when you `export PATH="/my/path/bin:$PATH"`, `PATH` will be interpolated with spaces instead of colons.